mysql scripts/table_creation/FullDB.sql

```diff
@@ -0,0 +1,172 @@
+
+DROP TABLE IF EXISTS UserPrivilege;
+DROP TABLE IF EXISTS Privilege;
+DROP TABLE IF EXISTS UserAbility;
+DROP TABLE IF EXISTS Ability;
+DROP TABLE IF EXISTS Login;
+DROP TABLE IF EXISTS PasswordReset;
+DROP TABLE IF EXISTS LoginHistory;
+DROP TABLE IF EXISTS User;
+
+CREATE TABLE LitmusQuestion (
+	litmusQuestionId INTEGER UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+    text VARCHAR(255) NOT NULL,
+	orderIndex INTEGER NOT NULL,
+    tag VARCHAR(255) NOT NULL,
+    required BOOLEAN NOT NULL DEFAULT TRUE,
+    archived BOOLEAN NOT NULL DEFAULT FALSE,
+    followUpTo INTEGER UNSIGNED NULL,
+    FOREIGN KEY(followUpTo) REFERENCES Litmusquestion(litmusQuestionId),
+	CONSTRAINT uniqueTag UNIQUE INDEX(tag)
+);
+
+CREATE TABLE PossibleAnswer (
+	possibleAnswerId INTEGER UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+    text VARCHAR(255) NOT NULL,
+    orderIndex INTEGER NOT NULL,
+    archived BOOLEAN NOT NULL DEFAULT FALSE,
+    litmusQuestionId INTEGER UNSIGNED NOT NULL,
+    FOREIGN KEY (litmusQuestionId) REFERENCES LitmusQuestion(litmusQuestionId)
+);
+
+CREATE TABLE Site (
+	siteId INTEGER UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+    title VARCHAR(255) NOT NULL,
+    address VARCHAR(255) NOT NULL,
+    phoneNumber VARCHAR(20) NOT NULL,
+    appointmentOnly BOOLEAN NOT NULL DEFAULT FALSE
+);
+
+CREATE TABLE Client (
+	clientId INTEGER UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+    firstName VARCHAR(255) NOT NULL,
+    lastName VARCHAR(255) NOT NULL,
+    emailAddress VARCHAR(255) NOT NULL
+);
+
+CREATE TABLE Appointment (
+	appointmentId INTEGER UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+    scheduledTime DATETIME NOT NULL,
+	createdAt TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+    archived BOOLEAN NOT NULL DEFAULT FALSE,
+	clientId INTEGER UNSIGNED NOT NULL,
+    FOREIGN KEY (clientId) REFERENCES Client(clientId),
+    siteId INTEGER UNSIGNED NOT NULL,
+    FOREIGN KEY (siteId) REFERENCES Site(siteId)
+);
+
+CREATE TABLE Answer (
+	answerId INTEGER UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+	possibleAnswerId INTEGER UNSIGNED NOT NULL,
+    FOREIGN KEY (possibleAnswerId) REFERENCES PossibleAnswer(possibleAnswerId),
+    appointmentId INTEGER UNSIGNED NOT NULL,
+    FOREIGN KEY (appointmentId) REFERENCES Appointment(appointmentId),
+    litmusQuestionId INTEGER UNSIGNED NOT NULL,
+    FOREIGN KEY (litmusQuestionId) REFERENCES LitmusQuestion(litmusQuestionId)
+);
+
+CREATE TABLE User (
+	userId INTEGER UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+	firstName VARCHAR(65) NULL,
+	lastName VARCHAR(65) NULL,
+	email VARCHAR(355) NOT NULL,
+	phoneNumber VARCHAR(20) NULL,
+	preparesTaxes BOOLEAN NOT NULL DEFAULT FALSE, 
+	archived BOOLEAN NOT NULL DEFAULT FALSE
+);
+
+
+CREATE TABLE ServicedAppointment (
+	servicedAppointmentId INTEGER UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+    startTime DATETIME,
+    endTime DATETIME,
+    createdAt TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
```

Comment:
A createdAt records when the record was actually inserted into the database. So no, this is not redundant since an Appointment createdAt would be set when the client makes an appointment. And the servicedAppointment createdAt would be set when a volunteer starts with a client. That being said, it is potentially slightly redundant since it would be created at either the startTime or the endTime in reality. Will change. And actually, on a side note, I'll look through each table and decide how much of an audit trail we need on each.